vdaf/prio3/arith/arith.go

```diff
@@ -0,0 +1,133 @@
+//go:generate go run gen.go
+
+// Package arith provides arithmetic operations over prime fields, vectors,
+// and polynomials.
+package arith
+
+import (
+	"encoding"
+	"io"
+
+	"github.com/cloudflare/circl/internal/conv"
+	"github.com/cloudflare/circl/internal/sha3"
+	"golang.org/x/crypto/cryptobyte"
+)
+
+// Elt is any type that stores a prime field element.
+type Elt any
+
+// Fp lists the functionality that a prime field element must have denoting
+// methods with a pointer receiver.
+type Fp[E Elt] interface {
```

Comment:
in theory they overlap, but in practice not all the crypto algorithms require the same interface for field elements.
This interface is written here just because generics need to specify it explicitly.

So far in CIRCL, each crypto algorithm defines its own set of methods for field elements.